tests/test.cpp

```diff
@@ -3222,6 +3222,23 @@ void FixedLengthArrayTest() {
   // Check alignment
   TEST_EQ(0, reinterpret_cast<uintptr_t>(mArStruct->d()) % 8);
   TEST_EQ(0, reinterpret_cast<uintptr_t>(mArStruct->f()) % 8);
+
+  // Check if default constructor set all memory zero
+  const size_t arr_size = sizeof(MyGame::Example::ArrayStruct);
+  char *non_zero_memory = new char[arr_size];
+  // set memory chunk of size ArrayStruct to 1's
+  std::memset(static_cast<void *>(non_zero_memory), 1, arr_size);
+  // after placement-new it should be all 0's
+  new (non_zero_memory) MyGame::Example::ArrayStruct;
+  bool have_non_zeroes = false;
+  for (size_t i = 0; i < arr_size; ++i) {
+    if (non_zero_memory[i] != 0) {
```

Comment:
just put the `TEST_EQ` in here, that way you don't need the boolean or the break, since this will terminate anyway on false.

Reply:
actually i was worried about performance here :) like TEST_EQ looks much heavy than just boolean

but you are right for sure - performance optimizations aren't that necessary in tests

Reply:
done
https://github.com/google/flatbuffers/pull/5938/commits/90d7534676bbb621f05bb89fa31b4be15e9e9ab5